src/os/Transaction.h

```diff
@@ -241,9 +241,17 @@ class Transaction {
   ceph::buffer::list data_bl;
   ceph::buffer::list op_bl;
 
-  std::list<Context *> on_applied;
-  std::list<Context *> on_commit;
-  std::list<Context *> on_applied_sync;
+  std::list<std::unique_ptr<Context>> on_applied;
```

Comment:
Maybe naming this type? and, then, possibly - making the 'transform' below part of its interface?

Reply:
Do you mean an alias / typedef?

Reply:
Either an alias, or (if additional functionality might make sense, like the suggested one above) - a structure